### PR TITLE
Build a mayastor-client image (#547)

### PR DESCRIPTION
**Pull request by bobek**
Ideas is to have a published image on dockerhub as desribed at #547 .

image

### CODE REVIEW

File: nix/pkgs/mayastor/default.nix [hunk 1]
```diff
@@ -132,4 +132,38 @@ in
       install * $out/bin
     '';
   };
+
+  # this image to simplify troubleshooting at existing clusters
+  client = stdenv.mkDerivation {
+    name = "mayastor-client";
+    inherit version;
+    src = [
+      ../../../target/debug/mayastor-client
```
Comment (tiagolobocastro) [inline nix/pkgs/mayastor/default.nix:141]: not sure this is going to work for CI... if it gets scheduled on a workspace were the debug build was not done

Comment (bobek) [reply nix/pkgs/mayastor/default.nix]: Thank you for the comment. I hope that this is solved by producing `mayastor-client` as overlay on the top of the `mayastor`.  So, "standard" way current.

**Comment by bobek** [comment]
I will investigate options for even smaller images, but we should be fine for now with basing the image off `mayastor`.

**Review by tiagolobocastro — APPROVED** [review]

**Review by blaisedias — APPROVED** [review]
LGTM

**Comment by bobek** [comment]
bors merge

**Pull request merged**